### PR TITLE
Add adapter for IProject > (bnd) Project

### PR DESCRIPTION
**Pull request by laeubi**
If one wants to write generic UI components then there is often the question to answer to what (bnd) project an Eclipse IProject belongs (if any).

This adds an Adapter that can adapt from IProject > (bnd) Project if the IProject is a PDE Bnd backed project.

See also
- https://bnd.discourse.group/t/announcement-bnd-and-pde-cooperation/372
- Corresponding bndtools change https://github.com/bndtools/bnd/pull/5885

### CODE REVIEW

File: ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/bnd/PdeBndAdapter.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+public class PdeBndAdapter implements IAdapterFactory {
+
+	@Override
+	public <T> T getAdapter(Object adaptableObject, Class<T> adapterType) {
+		if (adapterType == Project.class) {
+			if (adaptableObject instanceof IProject project) {
+				try {
+					return adapterType.cast(BndProjectManager.getBndProject(project).orElse(null));
+				} catch (Exception e) {
+					// can't adapt then...
```
Comment (szarnekow) [inline ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/bnd/PdeBndAdapter.java:34]: Should this be logged at least?

File: ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/bnd/PdeBndAdapter.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.IAdapterFactory;
+import org.osgi.service.component.annotations.Component;
+
+import aQute.bnd.build.Project;
+
+@Component(service = IAdapterFactory.class, property = {
+		IAdapterFactory.SERVICE_PROPERTY_ADAPTABLE_CLASS + "=org.eclipse.core.resources.IProject",
+		IAdapterFactory.SERVICE_PROPERTY_ADAPTABLE_CLASS + "=aQute.bnd.build.Project" })
```
Comment (HannesWell) [inline ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/bnd/PdeBndAdapter.java:24]: Isn't this an unidirectional adaption from IProject to (BND)-Project?
Therefore it does not look right that `SERVICE_PROPERTY_ADAPTABLE_CLASS` is used for both source and target type.
If I understood the doc correctly it should be:
```suggestion
		IAdapterFactory.SERVICE_PROPERTY_ADAPTER_NAMES + "=aQute.bnd.build.Project" })
```

Comment (HannesWell) [reply ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/bnd/PdeBndAdapter.java:24]: Looking at https://github.com/bndtools/bnd/pull/5885/files#diff-798b65a162b9e5e53ee378226f1a31b3c72511f2d12c6aec6bea5c31ced1d891 seems to confirm my assumption, nevertheless I have the impression that the implementation and service properties don't really provide the opposite but the same direction. Or was it not the intend to provide an (BND-project->IProject adoption there?)

File: ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/bnd/PdeBndAdapter.java (hunk 1 above)
Comment (laeubi) [reply ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/bnd/PdeBndAdapter.java:34]: The `Exception` is sadly a burden from the BND project that uses throws Exception all the time :-\
In this case it actually *should* never happen but if it does something is wrong and caller can just give up as we can't decide whats going wrong.

File: ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/bnd/PdeBndAdapter.java (hunk 2 above)
Comment (laeubi) [reply ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/bnd/PdeBndAdapter.java]: > Or was it not the intend to provide an (BND-project->IProject adoption there?)

It should be `IProject` -> (bnd) `Project`

**Pull request merged**